internal/pkg/runtime/engines/singularity/container_linux.go

```diff
@@ -540,6 +540,20 @@ func (c *container) mountGeneric(mnt *mount.Point) (err error) {
 		}
 	}
 	_, err = c.rpcOps.Mount(source, dest, mnt.Type, flags, optsString)
+	// when using user namespace we always try to apply mount flags with
+	// remount, then if we get a permission denied error, we continue
+	// execution by ignoring the error and warn user if the bind mount
+	// need to be mounted read-only
+	if remount && err != nil && c.userNS {
+		if err.Error() == syscall.Errno(syscall.EPERM).Error() {
```

Comment:
this is comparing the error strings. This should be done only as a last resort...

You can type-assert err (`err.(syscall.Errno)`) and then compare that with `syscall.EPERM`.

Reply:
It's the last resort, here `err` is of type `ServerError`